app/models/manageiq/providers/foreman/provider.rb

```diff
@@ -20,7 +20,7 @@ class ManageIQ::Providers::Foreman::Provider < ::Provider
 
   delegate :api_cached?, :ensure_api_cached, :to => :connect
 
-  before_create :ensure_managers
+  before_validation :ensure_managers
```

Comment:
@bdunne - I think it should be before_validation, in order for the validation to run with the new names.